### PR TITLE
fix(sync): handle reverted WoA sites

### PR DESCRIPTION
**Pull request by nightnei**
## Related issues

- Fixes https://github.com/Automattic/dotcom-forge/issues/9374

## Proposed Changes
There is a case where user connected WoA site with Business plan and then reverted it, so it becomes with errors as "Unsupported plan" or "Please enable hosting features".
With this PR we handle only reverting sites. Deleted sites will be handled in separate PR, to simplify review, since there is nuance - ATM [we don't load deleted sites](https://github.com/Automattic/studio/blob/trunk/src/hooks/use-fetch-wpcom-sites.tsx#L109), so necessary to think and maybe make some refactoring of existing code.

## Testing Instructions
1. Checkout to this branch and run `STUDIO_SITE_SYNC=true npm start`
2. Open wordpress.com and purchase two sites with Business plan
3. Open Studio and connect one of them
4. Open wordpress.com/me/purchases and cancel Business plan subscription for previously connected site
5. Open Studio and assert that you see error:<br />Screenshot 2024-11-20 at 12 09 07
6. Click on "Reconnect" button
7. Assert that you see dialog for connecting sites
8. Find your unsupported site
9. Asser that it has "Unsupported plan"<br />Screenshot 2024-11-20 at 12 10 56
10. Reset search ("x" button on the right side of search field)
11. Find the second site, which was bought on the second step in this Testing instructions, select and click "Conenct button"
12. Assert that after reconnecting the old connection disappeared and the new connection appeared into Sync tab

### CODE REVIEW

**Comment by nightnei** [comment]
@wojtekn you asked `What if we also linked 'Please enable hosting features' and 'Unsupported plan' to the same help page 'Get help' will point to?` inside the task. So I added this arrow. Does it look good for you or you would change something?
cc @matt-west 

Screenshot 2024-11-20 at 12 15 53

**Comment by nightnei** [comment]
@wojtekn I can't find information that is relevant to our "Get help" button link. Could you please point me?

File: src/components/sync-connected-sites.tsx [hunk 1]
```diff
@@ -146,141 +116,231 @@ export function SyncConnectedSites( {
...
+					{ section.name }
+					{ hasConnectionErrors && mainSite && ` (${ mainSite.url })` }
+				</div>
+				<Button
+					variant="link"
+					className="!ml-auto !text-a8c-gray-70 hover:!text-a8c-red-50 "
+					onClick={ handleDisconnectSite }
+					disabled={ isAnySitePulling || isAnySitePushing }
+				>
+					{ __( 'Disconnect1' ) }
```
Comment (wojtekn) [inline src/components/sync-connected-sites.tsx]: Somehow '1' slipped in to this label.

**Comment by nightnei** [comment]
@wojtekn @matt-west oh, one more thing that I made - by default all sites has title "Site title", so if there are two sites with errors - it's impossible to distinguish. Url is unique, so I think that we should render the URL, it's even more valuable then site title. WDYT about it and what do you think about UI?
Screenshot 2024-11-20 at 12 54 04

File: src/components/sync-connected-sites.tsx (hunk 1 above)
Comment (nightnei) [reply src/components/sync-connected-sites.tsx]: Yep, already reverted it 
Thanks Wojtek for thorough review 👍

**Comment by wojtekn** [comment]
In general, the flow works as described.

> Open wordpress.com and remove subscription for previously connected site

This instruction was unclear to me. I assumed that it was about reverting Atomic site, so I used a Bulk Delete Test Sites tool to do this. It allowed Studio to mark the site as an error.

> you asked What if we also linked 'Please enable hosting features' and 'Unsupported plan' to the same help page 'Get help' will point to? inside the task. So I added this arrow. Does it look good for you or you would change something?

Let's link the whole text, not the arrow itself. 

> I can't find information that is relevant to our "Get help" button link. Could you please point me?

I noticed that both results with Studio crashing due to an invalid link. Let's link those as proposed under https://github.com/Automattic/dotcom-forge/issues/9877, so:

- "Enable hosting features ↗" points to the page that allows enabling hosting features
- "Upgrade plan ↗" points to a plan comparison page that allows buying higher plan
- "Get help ↗" points to https://developer.wordpress.com/docs/developer-tools/studio/. We will adjust it later when we add a section for Sync.

**Comment by wojtekn** [comment]
> oh, one more thing that I made - by default all sites has title "Site title", so if there are two sites with errors - it's impossible to distinguish. Url is unique, so I think that we should render the URL, it's even more valuable then site title. WDYT about it and what do you think about UI?

I would skip the URL there. Connecting a Studio site to two external sites with the same name seems like an edge case.

**Comment by nightnei** [comment]
>This instruction was unclear to me

Got it. I will provide more details.

>I assumed that it was about reverting Atomic site, so I used a Bulk Delete Test Sites tool to do this. It allowed Studio to mark the site as an error.

It's also an option 👍
Btw, I wonder - is it possible to do revert in wordpress.com UI? I haven't found such opportunity.

>I noticed that both results with Studio crashing due to an invalid link. Let's link those as proposed under https://github.com/Automattic/dotcom-forge/issues/9877, so:

Added:<br /> Screenshot 2024-11-20 at 14 22 46

**Comment by wojtekn** [comment]
> Btw, I wonder - is it possible to do revert in wordpress.com UI? I haven't found such opportunity.

No, users who want to downgrade need to contact HE. See the P9HQHe-1sw-p2.

**Review by wojtekn — APPROVED** [review]

**Comment by matt-west** [comment]
Sorry for the delayed response.

I agree with @wojtekn that we don’t need to add the site URL alongside the title if there’s an error with the connection.

> you asked What if we also linked 'Please enable hosting features' and 'Unsupported plan' to the same help page 'Get help' will point to? inside the task. So I added this arrow. Does it look good for you or you would change something?

I was going to suggest making the full label the link, but it looks like you did that already. :)
